subprojects/matrix-benchmarking-workloads/rhods-notebooks-ux/plotting/multi_notebook_spawn_time.py

```diff
@@ -47,33 +50,27 @@ def do_plot(self, ordered_vars, settings, setting_lists, variables, cfg):
 
             if check_thresholds:
                 threshold_status_keys.add(entry_name)
-
-            for user_idx, ods_ci in entry.results.ods_ci.items() if entry.results.ods_ci else []:
-                accumulated_timelength = 0
-
-                if not ods_ci: continue
-                if not getattr(ods_ci, "output", False): continue
-
-                for step_name, test_times in ods_ci.output.items():
-                    if test_times.status != "PASS":
-                        continue
-
-                    timelength = (test_times.finish - test_times.start).total_seconds()
-
-                    accumulated_timelength += timelength
-                    if step_name != cfg__time_to_reach_step:
-                        continue
-
-                    break
-
-                thr90 = int(entry.results.thresholds.get("launch_time_90", 0)) or None
-                thr75 = int(entry.results.thresholds.get("launch_time_75", 0)) or None
-
-                data.append(dict(Version=entry_name,
-                                 SortIndex=sort_index,
-                                 LaunchTime90Threshold=thr90,
-                                 LaunchTime75Threshold=thr75,
-                                 Time=accumulated_timelength))
+            
+            accumulated_timelength = 0
+            current_index = -1
+            for user_idx, step_name, step_status, step_time in utils.parse_users(entry):
+                if current_index != user_idx:
+                    accumulated_timelength = 0
+                    current_index = user_idx
+
+                if step_status != "PASS":
+                    continue
+                
+                accumulated_timelength += step_time
+                if step_name == cfg__time_to_reach_step:
```

Comment:
```suggestion
                if step_name != cfg__time_to_reach_step:
                    continue
```
(and unindent the lines after)

--> always prefer writing code at the lower indentation level as possible
<--> the lowest indentation levels are the most important